linera-core/src/updater.rs

```diff
@@ -411,11 +462,13 @@ where
                     .read_certificate(*hash)
                     .await?
                     .ok_or_else(|| ChainClientError::MissingConfirmedBlock(*hash))?;
+                let admin_id = self.admin_chain_id().await?;
                 match self.send_confirmed_certificate(certificate, delivery).await {
                     Err(ChainClientError::RemoteNodeError(NodeError::EventsNotFound(
                         event_ids,
                     ))) if event_ids.iter().all(|event_id| {
-                        event_id.stream_id.application_id == GenericApplicationId::System
+                        event_id.stream_id == StreamId::system(EPOCH_STREAM_NAME)
```

Comment:
Why do you use `.all` and not `.any`?

Reply:
The only way processing a confirmed block can fail is due to missing committees. (It's like with messages: A _confirmed_ block doesn't fail, even if its received messages weren't seen yet.)

Reply:
Sorry, I meant the only way a confirmed block can fail with `EventsNotFound`.

Reply:
A node that returns any `EventsNotFound` containing any event that isn't from the admin chain here is faulty.

(It's legitimate to return other `EventsNotFound` errors when handling a _proposal_, but not when handling a confirmed block.)